test/unit/services/test_system.py

```diff
@@ -2,284 +2,374 @@
-            }
-        ),
-    )
-    assert updated_system == system
-
-
-def test_update_with_non_existent_id(test_helpers, system_repository_mock, system_service):
-    """
-    Test updating a System with a non-existent ID
-
-    Verify that the 'update' method properly handles the update of a System with a non-existent ID
-    """
-    # Mock `get` to not return a System
-    test_helpers.mock_get(system_repository_mock, None)
-
-    system_id = str(ObjectId())
-    with pytest.raises(MissingRecordError) as exc:
-        system_service.update(system_id, MagicMock())
-    system_repository_mock.update.assert_not_called()
-    assert str(exc.value) == f"No System found with ID: {system_id}"
-
-
-def test_delete(system_repository_mock, system_service):
-    """
-    Test deleting a System
-
-    Verify that the `delete` method properly handles the deletion of a System by ID
-    """
-    system_id = MagicMock()
-
-    system_service.delete(system_id)
-
-    system_repository_mock.delete.assert_called_once_with(system_id)
+    def mock_get_breadcrumbs(self):
+        """Mocks repo methods appropriately to test the 'get_breadcrumbs' service method"""
+
+        # Simply a return currently, so no need to use actual data
+        self._expected_breadcrumbs = MagicMock()
+        self.test_helpers.mock_get_breadcrumbs(self.mock_system_repository, self._expected_breadcrumbs)
+
+    def call_get_breadcrumbs(self, system_id: str):
+        """Calls the SystemService `get` method"""
+
+        self._obtained_system_id = system_id
+        self._obtained_breadcrumbs = self.system_service.get_breadcrumbs(system_id)
+
+    def check_get_breadcrumbs_success(self):
+        """Checks that a prior call to `call_get` worked as expected"""
+
+        self.mock_system_repository.get_breadcrumbs.assert_called_once_with(self._obtained_system_id)
+
+        assert self._obtained_breadcrumbs == self._expected_breadcrumbs
+
+
+class TestGetBreadcrumbs(GetBreadcrumbsDSL):
+    """Tests for getting the breadcrumbs of a System"""
+
+    def test_get_breadcrumbs(self):
+        """Test getting a System's breadcrumbs"""
+
+        self.mock_get_breadcrumbs()
+        self.call_get_breadcrumbs(str(ObjectId()))
+        self.check_get_breadcrumbs_success()
+
+
+class ListDSL(SystemServiceDSL):
+    """Base class for list tests"""
+
+    _parent_id_filter: str
+    _expected_systems: MagicMock
+    _obtained_systems: MagicMock
+
+    def mock_list(self):
+        """Mocks repo methods appropriately to test the 'list' service method"""
+
+        # Simply a return currently, so no need to use actual data
+        self._expected_systems = MagicMock()
+        self.test_helpers.mock_list(self.mock_system_repository, self._expected_systems)
+
+    def call_list(self, parent_id: Optional[str]):
+        """Calls the SystemService `list` method"""
+
+        self._parent_id_filter = parent_id
+        self._obtained_systems = self.system_service.list(parent_id)
```

Comment:
It's complaining about the type - `Expected type 'MagicMock', got 'list[SystemOut]' instead`